tanjun/components.py

```diff
@@ -713,3 +724,20 @@ def _load_from_properties(self) -> None:
             if isinstance(member, LoadableProtocol):
                 if result := member.copy().load_into_component(self):
                     setattr(self, name, result)
+
+    def with_repeater(
+        self,
+        delay: typing.Union[datetime.timedelta, int, float],
+        *,
+        max_runs: typing.Optional[int] = None,
+        event_loop: typing.Optional[asyncio.AbstractEventLoop] = None,
+    ) -> collections.Callable[[CallbackSigT], abc.AbstractRepeater]:
```

Comment:
This is off standard for the lib, with_repeater should just take one positional arg of type typeVar(AbstractRepeater) and return it after calling self.add_repeater(repeater) to allow for custom impls